### PR TITLE
74043 API add mode id and responsible code to step endpoint

### PR DESCRIPTION
**Pull request by demissie**

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public Step(string plant, string title, Mode mode, Responsible responsible)
         public int ModeId { get; private set; }
         public int ResponsibleId { get; private set; }
 
-        public int SortKey { get; set; }  // sortKey will be set correct in later PBI when impl UI for Add, MoveUp and MoveDown of Steps in Journey
+        public int SortKey { get; set; }  // sortKey will be set correct in later PBI when impl UI for Add, MoveUp and MoveDown (swap) of Steps in Journey
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs:57]: remove the whole comment

File: src/Equinor.Procosys.Preservation.Command/JourneyCommands/UpdateStep/UpdateStepCommandValidator.cs [hunk 2]
```diff
@@ -10,19 +12,25 @@ public class UpdateStepCommandValidator : AbstractValidator<UpdateStepCommand>
     {
         public UpdateStepCommandValidator(
             IJourneyValidator journeyValidator,
-            IStepValidator stepValidator)
+            IStepValidator stepValidator,
+            IModeValidator modeValidator,
+            IResponsibleValidator responsibleValidator)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/JourneyCommands/UpdateStep/UpdateStepCommandValidator.cs:17]: remove .. not in use

Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Command/JourneyCommands/UpdateStep/UpdateStepCommandValidator.cs:17]: responsibleValidator is not in use

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs [hunk 3]
```diff
@@ -86,5 +86,25 @@ public void SetModified(Person modifiedBy)
 
             ModifiedById = modifiedBy.Id;
         }
+
+        public void SetMode(Mode mode)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs:90]: Add a // Todo-comment here: Add test in StepTests.cs (implment in next PR)

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs [hunk 4]
```diff
@@ -86,5 +86,25 @@ public void SetModified(Person modifiedBy)
...
+        {
+            if (mode == null)
+            {
+                throw new ArgumentNullException(nameof(mode));
+            }
+
+            ModeId = mode.Id;
+        }
+
+        public void SetResponsible(Responsible responsible)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/JourneyAggregate/Step.cs:100]: Add a // Todo-comment here: Add test in StepTests.cs (implement in next PR)

File: src/Equinor.Procosys.Preservation.Infrastructure/Repositories/ResponsibleRepository.cs [hunk 5]
```diff
@@ -7,5 +9,8 @@ public class ResponsibleRepository : RepositoryBase<Responsible>, IResponsibleRe
         public ResponsibleRepository(PreservationContext context) : base(context.Responsibles)
         {
         }
+
+        public Task<Responsible> GetByCodeAsync(string responsibleCode)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Infrastructure/Repositories/ResponsibleRepository.cs:13]: Add a // Todo-comment here: Add test in ResponsibleRepositoryTests.cs (implement in next PR)

File: src/Equinor.Procosys.Preservation.Query/JourneyAggregate/GetAllJourneysQueryHandler.cs [hunk 6]
```diff
@@ -55,7 +55,12 @@ where responsibleIds.Contains(r.Id)
                                     .Select(r => new ResponsibleDto(r.Id, r.Code, r.Title,
                                         r.RowVersion.ConvertToString()))
                                     .Single();
-                                return new StepDto(s.Id, s.Title, s.IsVoided, modeDto, responsibleDto);
+                                return new StepDto(
+                                    s.Id,
+                                    s.Title,
+                                    s.IsVoided,
+                                    s.RowVersion.ConvertToString(),
+                                    modeDto, responsibleDto);
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Query/JourneyAggregate/GetAllJourneysQueryHandler.cs:63]: RowVersion should be the last in the list

File: src/Equinor.Procosys.Preservation.Query/JourneyAggregate/GetJourneyByIdQueryHandler.cs [hunk 7]
```diff
@@ -50,6 +50,7 @@ where responsibleIds.Contains(r.Id)
                         step.Id,
                         step.Title,
                         step.IsVoided,
+                        step.RowVersion.ConvertToString(),
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Query/JourneyAggregate/GetJourneyByIdQueryHandler.cs:53]: RowVersion should be last in the list

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs [hunk 8]
```diff
@@ -13,10 +13,19 @@ namespace Equinor.Procosys.Preservation.Command.Tests.JourneyCommands.UpdateStep
...
         private readonly string _rowVersion = "AAAAAAAAABA=";
 
         private readonly string _oldTitle = "StepTitleOld";
         private readonly string _newTitle = "StepTitleNew";
+        private Mock<Mode> _modeMock;
+        private Mock<Responsible> _responsibleMock;
+        private Mock<IModeRepository> _modeRepositoryMock;
+        private Mock<IResponsibleRepository> _responsibleRepositoryMock;
+        private Mock<IPlantProvider> _plantProviderMock;
+
```
Comment (llykkelarsen) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs]: remove whitespace

File: src/Equinor.Procosys.Preservation.Command/JourneyCommands/UpdateStep/UpdateStepCommandValidator.cs (hunk 2 above)
Comment (llykkelarsen) [reply src/Equinor.Procosys.Preservation.Command/JourneyCommands/UpdateStep/UpdateStepCommandValidator.cs:17]: I see that Jan Inge said the same thing so I will resolve this :)

**Pull request merged**